src/datapoint.py

```diff
@@ -0,0 +1,114 @@
+from typing import Dict
+import numpy as np
+from enum import Enum
+from abc import ABC
+
+
+class Method(Enum):
```

Comment:
This should be a string enum.
```suggestion
class Method(Enum, str):
```